x/gasless/spec/01_concepts.md

```diff
@@ -0,0 +1,143 @@
+<!--
+order: 1
+-->
+
+# Concepts
+
+## Gasless
+
+The Gasless module provides functionality for Developers (Native Go and Smart Contract) to cover the execution fees of transactions interacting with their contracts or Messages. This aims to improve the user experience and help onboard wallets with little to no available funds.
+Developers can setup their MessageTypes and Contracts with Gasless by first creating a GasTank and whitelisting MessageTypes and ContractAddresses with preferred configurations and then funding it with chain's native fee token.
+Clients can then interact with the Messages and Contract normally as usual and the fee for the tx will be deducted from the gastank
+
+### Creating a GasTank
+
+There can be multiple gastanks, each of which can be with unique configuration.
+
+```console
+foo@bar:~$ aibd tx gasless create-gas-tank [fee-denom] [max-fee-usage-per-tx] [max-txs-count-per-consumer] [max-fee-usage-per-consumer] [txs-allowed] [contracts-allowed] [gas-deposit]
+```
+
+e.g
+
+```console
+foo@bar:~$ aibd tx gasless create-gas-tank aaib 2000000 100 200000000 "/cosmos.bank.v1beta1.MsgMultiSend,/cosmos.bank.v1beta1.MsgSend" "rol14hj2tavq8f....,rol14hj2t...." 100000000aaib --from cooluser --chain-id test-1
+```
+
+In the above tx -
+
+- `fee-denom` - all txs with this fee-denom can consume gas from this gas tank
+- `max-fee-usage-per-tx` - maximum fee a tx can utilize, if the asked tx fee > max-fee-usage-per-tx, then the fee will be deducted for the tx maker.
+- `max-txs-count-per-consumer` - no. of txs allowed for each address
+- `max-fee-usage-per-consumer` - max fee usage for each address
+- `txs-allowed` - list of message types, which are to be whitelisted for gasless tx
+- `contracts-allowed` - list of contract addresses, which are to be whitelisted for gasless tx
+- `gas-deposit` - initial deposit for the gastank
+
+> Note : anyone can create the gastank and whitelist the contract address or message type of their own will
+
+### Authorizing Actors
+
+The creator or owner of the gastank can allow other trusted addresses to manage their gas tanks, these actors can block and unblock the specific address/consumers txs going through this tank.
+
+```console
+foo@bar:~$ aibd tx gasless update-authorized-actors [gas-tank-id] [actors]
+```
+
+e.g
+
+```console
+foo@bar:~$ aibd tx gasless update-authorized-actors 1 "rol14hj2tavq8f....,rol14hj2t...." 100000000aaib --from cooluser --chain-id test-1
+```
+
+### Updating GasTank Status
+
+A GasTank can be disabled and enabled by owner anytime
+
+```console
+foo@bar:~$ aibd tx gasless update-gas-tank-status [gas-tank-id]
+```
+
+e.g
+
+```console
+foo@bar:~$ aibd tx gasless update-gas-tank-status 1 --from cooluser --chain-id test-1
```

Comment:
the current status of the gas tank can be fetched with available query - 
`aibd  q gasless gastank [gas-tank-id]`,

So whenever the tx is made, the tank will be updated with the opposite status i.e 
if it is active - after tx - it will be inactive
and vice versa